core/src/main/java/org/elasticsearch/action/count/CountRequest.java

```diff
@@ -64,12 +62,12 @@
     @Nullable
     private String preference;
 
-    private BytesReference source;
-
     private String[] types = Strings.EMPTY_ARRAY;
 
     private int terminateAfter = DEFAULT_TERMINATE_AFTER;
 
+    private SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
```

Comment:
one other note: using SearchSourceBuilder internally means that all the options that the search api supports are exposed to the count api too, via java api. But the count api only supports a small subset of those features. Are we sure we don't want to keep on using the QuerySourceBuilder?